torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -418,7 +421,8 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
     {
       std::unique_lock<std::mutex> lock(timeoutMapMutex_);
       auto& timeoutFuturesVector = timeoutMap_[expirationTime];
-      timeoutFuturesVector.emplace_back(futureResponseMessage);
+      timeoutFuturesVector.emplace_back(
+          std::make_pair(futureResponseMessage, timeout));
```

Comment:
I think you might not need to call `std::make_pair` with `emplace_back`?

Reply:
Great catch!